doc/source/ray-job-submission/overview.rst

```diff
@@ -0,0 +1,239 @@
+
+.. note::
+  **The goal of Ray job submssion is to provide a lightweight mechanism for user to submit their locally developed and tested application to a running local / remote ray cluster, thus enabling the user to package, deploy, and manage their ray application as jobs. These jobs can be submitted by a job manager of their choice.**
+
+Example - Setup
+---------------
+
+Let's start with a sample ray script as example for job submission. Once executed locally, this script will use ray APIs to print counter value of a remote actor from 1 to 5, and print the version of 'requests' module it's using.
+
+We can put this file in a local directory of your choice, with filename "script.py", so your working directory will look like:
+
+.. code-block:: bash
+
+  | your_working_directory ("./")
+  | ├── script.py
+
+.. code-block:: python
+
+    import ray
+    import requests
+
+    ray.init()
+
+    @ray.remote
+    class Counter:
+        def __init__(self):
+            self.counter = 0
+
+        def inc(self):
+            self.counter += 1
+
+        def get_counter(self):
+            return self.counter
+
+    counter = Counter.remote()
+
+    for _ in range(5):
+        ray.get(counter.inc.remote())
+        print(ray.get(counter.get_counter.remote()))
+
+    print(requests.__version__)
+
+Ray Job SDK
+------------
+
+Ray job sdk is the recommended way to submit jobs programmatically.
+
+| Start a local ray cluster headnode
+
+.. code-block:: bash
+
+    ray start --head
+
+We can import and intialize job submission client by providing an valid ray cluster headnode address where port is same as ray dashboard. We're using your local ray cluster as example but it works the same for remote ray cluster addresses.
+
+.. code-block:: python
+
+    from ray.dashboard.modules.job.sdk import JobSubmissionClient
+
+    client = JobSubmissionClient("http://127.0.0.1:8265")
+
+Then we can submit our application to ray cluster via job SDK.
+
+.. code-block:: python
+
+    job_id = client.submit_job(
+        # Entry point to execute
+        entrypoint="python script.py",
+        # Working dir
+        runtime_env={
+            "working_dir": "./",
+            "pip": ["requests==2.26.0"]
+        }
+    )
+
+.. tip::
+
+    By default ray job server will generate a new uuid as return value, but you can also generate your unique job_id first and pass it into :code:`submit_job`. In this case the job will be executed with your given id, and will throw error if same job_id is submitted more than once for the same ray cluster. This can facilitate job id management assuming you have your own job registry to generate and persist unique job ids.
+
+Now we can have a simple polling loop that checks job status until it reaches terminal state, and get logs at the end. We are expected to see actor printed numbers as well as correct version of :code:`requests` module used via runtime_env.
```

Comment:
```suggestion
Now we can have a simple polling loop that checks the job status until it reaches its terminal state, and gets the logs at the end. We expect to see actor printed numbers as well as the correct version of the :code:`requests` module specified in the ``runtime_env``.
```

Reply:
What's "actor printed numbers"?  Also, is it possible to paste the expected output?